src/main/java/edu/harvard/iq/dataverse/engine/command/impl/FinalizeDatasetPublicationCommand.java

```diff
@@ -83,9 +83,20 @@ public Dataset execute(CommandContext ctxt) throws CommandException {
             validateDataFiles(theDataset, ctxt);
             // (this will throw a CommandException if it fails)
         }
-        
+
+		/*
+		 * Try to register the dataset identifier (making it public). The
+		 * registerExternalIdentifier command will also create the identifier if needed:
+		 * - In the case where the PID provider doesn't support reservations, in which
+		 *   case it will retry if collisions are found 
+		 * - In the case where the PID Provider
+		 *   does support reservations, but the Dataset's PID hasn't been reserved (was
+		 *   created prior to v5.0 and admin hasn't run reserve API, or PID Provider was
+		 *   offline at creation time), in which case no retries will be attempted.
+		 */
+        GlobalIdServiceBean idServiceBean = GlobalIdServiceBean.getBean(getDataset().getProtocol(), ctxt);
         if ( theDataset.getGlobalIdCreateTime() == null ) {
-            registerExternalIdentifier(theDataset, ctxt);
+            registerExternalIdentifier(theDataset, ctxt, idServiceBean.registerWhenPublished());
```

Comment:
Independent of the larger discussion about the PR, if we keep the new boolean flag, it could always be set to false here - PID providers that do not register draft DOIs would then try once to create a DOI at publication time and fail if the DOI is already in use, versus retrying and causing all of the files to be 'lost' (at the path for the old/conflicted DOI instead of one found in a retry). Retries would then only occur in the create command (for providers that can reserve).

Reply:
This does make sense - maybe we should just hard-code it to false, when the method is called from the publication command. It's not exactly clear what the point would be, in keeping the current behavior for handles. 
?